Core/Core/Extensions/NSPersistentContainerExtensions.swift

```diff
@@ -84,4 +84,23 @@ extension NSPersistentContainer {
             print(error)
         }
     }
+
+    @objc open func performWriteTask(_ block: @escaping (NSManagedObjectContext) -> Void) {
```

Comment:
The Enrollment <-> Submission relationship with both exhausting in SubmissionList led to sometimes they would get edited & deleted (due to always deleting all objects on first write of paginated data) in separate contexts and then they can't merge properly.

Instead of having every use case create 2 new background contexts to do its thing, always write to a single saved background context.